protocol.md

```diff
@@ -1,55 +1,248 @@
+Once an exchange flow is started, the subcommand cannot be changed.
+
+| Name        | Value | Description                                                                                     |
+| ----------- | ----- | ----------------------------------------------------------------------------------------------- |
+| SWAP_LEGACY | 0x00  | SWAP transaction (crypto against crypto) using legacy signature and encoding, deprecated soon   |
+| SELL_LEGACY | 0x01  | SELL transaction (crypto against fiat) using legacy signature and encoding, deprecated soon     |
+| FUND_LEGACY | 0x02  | FUND transaction (crypto to owned account) using legacy signature and encoding, deprecated soon |
+| SWAP_NEW    | 0x03  | SWAP transaction (crypto against crypto) using unified signature and encoding                   |
+| SELL_NEW    | 0x04  | SELL transaction (crypto against fiat) using unified signature and encoding                     |
+| FUND_NEW    | 0x05  | FUND transaction (crypto to owned account) using unified signature and encoding                 |
+
+
+### EXTENSION:
+
+The maximum data in a single APDU is 255 bytes, in case it is not sufficient for a command, it is possible to send the commad in several parts. \
+It is given to the app as the upper part of P2 of an APDU. \
+It is only possible to use this field for a PROCESS_TRANSACTION_RESPONSE command in a SWAP_NEW, SELL_NEW, or FUND_NEW flow.
+
+| Name      | Value     | Description                                                       |
+| --------- | --------- | ----------------------------------------------------------------- |
+| P2_NONE   | 0x00 << 4 | This APDU is self sufficient                                      |
+| P2_EXTEND | 0x01 << 4 | This APDU is not whole, it extends a previously received one      |
+| P2_MORE   | 0x02 << 4 | This APDU is not whole, the followup part while be received later |
+
+### DATA:
+
+For all COMMANDS, the data bytes must be prefixed by a data length header. \
+For PROCESS_TRANSACTION_RESPONSE command of a SWAP_NEW, SELL_NEW, and FUND_NEW, the data length header is coded on 2 bytes. \
+For all other COMMANDS and all other TYPES, the data length header is coded on 1 byte.
+
+#### GET_VERSION
+
+No data expected.
+
+#### START_NEW_TRANSACTION
+
+No data expected.
+
+#### SET_PARTNER_KEY
+
+| Bytes   | Description                        |
+| ------- | ---------------------------------- |
+| 1 byte  | Length of the encoded partner name |
+| N bytes | Partner name encoded with utf-8    |
+| N bytes | Partner public key                 |
+
+#### CHECK_PARTNER
+
+| Bytes  | Description                                                                             |
+| ------ | --------------------------------------------------------------------------------------- |
+| N byte | Signature of the partner credentials by the Ledger key, curve secp256k1 hashfunc sha256 |
```

Comment:
```suggestion
| SIZE bytes | Signature of the partner credentials by the Ledger key, curve secp256k1 hashfunc sha256 |
```
or
```suggestion
| 1 byte  | Length of the signature of the partner credentials N                                          |
| B bytes | Signature of the partner credentials by the Ledger key, curve secp256k1 hashfunc sha256 |
```

Reply:
No length header field, only LC, added this info in the doc